File: TwoStreamModel-2.4/rad2sf90/gammaproc.f
! ---------------------------------------------------
! File            : gammaproc.f
! Author          : Lavergne Thomas                          
! Creation Date   : 11 August 2005                            
! Purpose         :                                  
!     Isolate the low-level procedures for the gamma coefficients
!     calculations.
! ---------------------------------------------------

MODULE gammaproc

IMPLICIT NONE

PRIVATE

PUBLIC :: gammas


CONTAINS

SUBROUTINE gammas(rl,tl,mu0,gammaCoeff)
   REAL(8), INTENT(IN) :: rl,tl,mu0
   REAL(8), DIMENSION(:), INTENT(OUT) :: gammaCoeff 

   REAL(8) :: wd,w0,w0half,wdsixth

   w0      = rl+tl
   wd      = rl-tl
   w0half  = w0*0.5_8
   wdsixth = wd/6.0_8

   gammaCoeff(1)=2.*(1. - w0half + wdsixth)
   gammaCoeff(2)=2.*(w0half + wdsixth)
   gammaCoeff(3)=2.*(w0half*0.5 + mu0*wdsixth)/w0
   gammaCoeff(4)=1.-gammaCoeff(3)
   
END SUBROUTINE gammas

END MODULE gammaproc

File: TwoStreamModel-2.4/rad2s/gammaproc.c
/* ----------------------------------------------------------------------
 * File            : gammaproc.c                       
 * Author          : Lavergne Thomas                          
 * Creation Date   : 09 August 2005                            
 * Purpose         :                                  
 *         Isolate the low-level procedures for the gamma coefficients
 *         calculations.
 * -----------------------------------------------------------------------
 */

#include "gammaproc.h"


void gammas(double rl,double tl,double mu0,double *gamma) 
{
   double wd  = rl - tl;
   double w0  = rl + tl;
   double w0half  = w0*0.5;
   double wdsixth = wd*RTSIXTH;
   
   gamma[0] = 2.*(1. - w0half + wdsixth);
   gamma[1] = 2.*(w0half + wdsixth);
   gamma[2] = 2.*(w0half*0.5 + mu0*wdsixth)/w0;
   gamma[3] = 1. - gamma[2];
}

